frontend/biology4u/src/components/TopicTest.vue

```diff
@@ -0,0 +1,241 @@
+                            <label style="width: 1rem">
+                                <input type="radio" :name="currentQuestion.id" :value="option" v-model="selectedOption">
+                                {{ option }}
+                            </label>
+                        </li>
+                    </ul>
+                    <div id="buttonsContainer">
+                        <button @click="prevQuestion">Предишен</button>
+                        <button @click="checkAnswer">{{ buttonText }}</button>
+                    </div>
+                </div>
+                <div v-else>
+                    <h2>Резултати от теста</h2>
+                    <p>Имаш {{ score }} верни въпроса от {{ questions.length }}.</p>
+                    <p>Време: {{ timeTaken }} секунди.</p>
+                    <p>Точки: {{ Math.ceil(finalScore) }}</p>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import TestService from "@/services/testResult.service";
+import QuestionService from "../services/question.service";
+import { mapGetters } from "vuex";
+export default {
+    data() {
+        return {
+            questions: [],
+            scores: [],
+            userAnswers: [],
+            selectedOptions: [],
+            currentQuestionIndex: 0,
+            selectedOption: null,
+            score: 0,
+            startTime: null,
+            endTime: null,
+            maxTime: null,
+            finished: false,
+            started: false,
+            table: true
+        };
+    },
+    computed: {
+        ...mapGetters({
+            user: "getUserInfo",
+        }),
+        currentQuestion() {
+            return this.questions[this.currentQuestionIndex];
+        },
+        buttonText() {
+            return this.currentQuestionIndex === this.questions.length - 1 ? "Предай" : "Следващ";
+        },
+        timeTaken() {
+            if (!this.startTime && !this.endTime) {
+                return 0;
+            }
+            const timeDiff = this.endTime - this.startTime;
+            const seconds = Math.round(timeDiff / 1000);
+            return seconds;
+        },
+    },
+
+    created() {
+        this.$watch(
+            () => this.$route.params.title,
+            () => {
+                this.questions = [];
+                this.currentQuestionIndex = 0;
+                this.selectedOption = null;
+                this.score = 0;
+                this.finalScore = 0;
+                this.startTime = null;
+                this.endTime = null;
+                this.finished = false;
+                this.started = false;
+                this.table = true;
+                this.getQuestions(this.$route.params.title);
+            },
+            { immediate: true }
+        )
+    },
+
+    beforeMount() {
+        // this.getQuestions(this.$route.params.title)
+    },
+
+    mounted() {
+        this.getScores(this.$route.params.title)
+    },
+
+    methods: {

```

Comment:
A lot of the methods of this class and SubTopicTest class are the same. In this case, you can either make a base class and extend it or extract the repeated functions and use it in both places. We want to be as [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) as possible.